src/components/bulk-email-tool/bulk-email-task-manager/BulkEmailContentHistory.jsx

```diff
@@ -1,41 +1,241 @@
+      setShowHistoricalEmailContentTable(true);
+    }
+  }
+
+  /**
+   * This function is responsible for setting the current `messageContent` state data. This will be the contents of a
+   * previously sent email message from the bulk course email tool. This also toggles a modal to be visible to display
+   * the message contents to the end user.
+   */
+  const onViewMessageClick = (tableData) => {
+    setMessageContent(tableData);
+    setIsMessageModalOpen(true);
+  };
+
+  /**
+   * Render function for the email content history table. If an error occurs while attempting to fetch data from
+   * edx-platform we will render this error instead of the table.
+   */
+  const renderError = () => (
+    <div>
+      <Alert variant="danger">
+        <p className="font-weight-bold">
+          {intl.formatMessage(messages.errorFetchingData)}
+        </p>
+      </Alert>
+    </div>
+  );
+
+  /**
+   * Render function for the email content history table. If there is no data to display in our table we will render
+   * this informative message instead.
+   */
+  const renderEmpty = () => (
+    <div className="pt-1">
+      <Alert variant="warning">
+        <p className="font-weight-bold">
+          {intl.formatMessage(messages.noEmailData)}
+        </p>
+      </Alert>
+    </div>
+  );
+
+  /**
+   * Renders a modal that will display the contents of a single historical email message sent via the bulk course email
+   * tool to a user.
+   */
+  const renderMessageModal = () => (
+    <div>
+      <Modal
+        open={isMessageModalOpen}
+        title=""
+        body={(
+          <div>
+            <div className="d-flex flex-row">
+              <p>
+                {intl.formatMessage(messages.modalMessageSubject)}
+              </p>
+              <p className="pl-2">
+                {messageContent.subject}
+              </p>
+            </div>
+            <div className="d-flex flex-row">
+              <p>
+                {intl.formatMessage(messages.modalMessageSentBy)}
+              </p>
+              <p className="pl-2">
+                {messageContent.requester}
+              </p>
+            </div>
+            <div className="d-flex flex-row">
+              <p>
+                {intl.formatMessage(messages.modalMessageTimeSent)}
+              </p>
+              <p className="pl-2">
+                {messageContent.created}
+              </p>
+            </div>
+            <div className="d-flex flex-row">
+              <p>
+                {intl.formatMessage(messages.modalMessageSentTo)}
+              </p>
+              <p className="pl-2">
+                {messageContent.sent_to}
+              </p>
+            </div>
+            <hr className="py-2" />
+            <div>
+              <p>
+                {intl.formatMessage(messages.modalMessageBody)}
+              </p>
+              <div dangerouslySetInnerHTML={{ __html: messageContent.email.html_message }} />
```

Comment:
WOW. What is this?

Reply:
This is to display the message HTML that we received from our API call. Without doing this, it shows the text with the markup ("<p>message</p>" rather than "message" (which we don't do today).

Reply:
This is unfortunately the only way to do this. That's why the prop is named that way. If we want to keep the same features we have on the instructor tool, this is the way to do it.